### PR TITLE
Fix problem with some software (like MSD.EXE) displaying garbage mouse driver info

### PR DESCRIPTION
**Pull request by FeralChild64**
Implements 3 functions returning mouse driver information - prevents some software from displaying garbage, like _Path to MOUSE.INI_ on screenshot below:

msd_000

### CODE REVIEW

**Comment by johnnovak** [comment]
I vote on leaving that smiley face alone! It's cute! 😂 

image

File: include/math_utils.h [hunk 1]
```diff
@@ -205,6 +205,16 @@ constexpr uint32_t clamp_to_uint32(const T val)
 	return static_cast<uint32_t>(std::clamp(val, min_val, max_val));
 }
 
+constexpr uint8_t read_low_nibble(const uint8_t byte)
+{
+	return static_cast<uint8_t>(byte % 0x10);
```
Comment (johnnovak) [inline include/math_utils.h:210]: `byte && 0x0f` and `(byte && 0xf0) >> 8`?
That's what I expect to see when working with nibbles, this gave me a pause.

File: src/hardware/input/mouseif_dos_driver.cpp [hunk 2]
```diff
@@ -49,6 +50,14 @@ CHECK_NARROWING();
 // - https://www.stanislavs.org/helppc/int_33.html
 // - http://www2.ift.ulaval.ca/~marchand/ift17583/dosints.pdf
 // - https://github.com/FDOS/mouse/blob/master/int33.lst
+// - https://www.fysnet.net/faq.htm
+
+// Versions are stored in BCD code - 0x09 = version 9, 0x10 = version 10, etc.
+static constexpr uint8_t driver_version_major = 0x08;
+static constexpr uint8_t driver_version_minor = 0x05;
+
+static const std::string str_version_prefix = "version ";
```
Comment (johnnovak) [inline src/hardware/input/mouseif_dos_driver.cpp:59]: The space is intentional? Maybe put a comment there that the trailing space is there for a reason (one could mistake it for a typo and remove it).

File: src/hardware/input/mouseif_dos_driver.cpp [hunk 3]
```diff
@@ -1611,6 +1629,24 @@ static Bitu int33_handler()
...
+		//     offset 0x06 - interrupt rate
+		//     offset 0x07 - cursor mask
+		//     offset 0x08 - laptop adjustment
+		//     offset 0x09 - memory type
+		//     offset 0x0a - super VGA flag
+		//     offset 0x0b - rotation angle (2 bytes)
+		//     offset 0x0d - primary button
+		//     offset 0x0e - secondary button
+		//     offset 0x0f - click lock enabled
+		//     offset 0x10 - acceleration curves tables (324 bytes)
```
Comment (johnnovak) [inline src/hardware/input/mouseif_dos_driver.cpp:1649]: Nice documentation 👍🏻

File: src/hardware/input/mouseif_dos_driver.cpp [hunk 4]
```diff
@@ -1742,6 +1782,60 @@ static Bitu user_callback_handler()
...
+
+	constexpr uint8_t bytes_per_block = 0x10;
+	auto length_blocks = static_cast<uint16_t>(length_bytes / bytes_per_block);
+	if (length_bytes % bytes_per_block) {
+		length_blocks = static_cast<uint16_t>(length_blocks + 1);
+	}
+
+	info_segment = DOS_GetMemory(length_blocks);
+
+	// TODO: if 'MOUSE.INI' file gets implemented, INT 33 functuion 0x32
```
Comment (johnnovak) [inline src/hardware/input/mouseif_dos_driver.cpp:1825]: typo: function

**Review by johnnovak — APPROVED** [review]
Nice one @FeralChild64 👍🏻

File: src/hardware/input/mouseif_dos_driver.cpp (hunk 4 above)
Comment (FeralChild64) [reply src/hardware/input/mouseif_dos_driver.cpp]: OK, fixed.

File: src/hardware/input/mouseif_dos_driver.cpp (hunk 2 above)
Comment (FeralChild64) [reply src/hardware/input/mouseif_dos_driver.cpp]: I got rid of this helper string, it is used in one place nevertheless.

File: include/math_utils.h (hunk 1 above)
Comment (FeralChild64) [reply include/math_utils.h]: For higher nibble it is `byte >> 4`, actually. OK, changed - due to past experiences with mathematics I'm more used to division and remainder operators, but I admin I'm an exception :)

Comment (johnnovak) [reply include/math_utils.h]: Ah yeah, `>> 4` 😄  Well, I'm used to seeing boolean ops and bit shifts everywhere in low-level code from my asm coding days, and while it does not matter much these days performance-wise, division/modulo was very heavy compared to bit-ops back in the day. So yeah, partly tradition, partly familiarity for me 😄 But maybe performance still matters, division is never really "lightweight" on any platform.

**Comment by kcgen** [comment]
Additions look great, @FeralChild64! Passing sanitizer and MSD tests.

msd_000

Another one down :-)

**Pull request merged**